src/test/java/wromaciej/hvac_sim/thermo/fluids/RealProcessTest.java

```diff
@@ -19,22 +20,24 @@
 
 public class RealProcessTest {
 
-    static {
-        //load library
-        FluidData.loadLibrary();
-    }
+
 
     @Test
     public void shouldReturnSameEndPointForIdealCompressionAndRealWith100PercentEfficiency(){
         //GIVEN
+        FluidData fluidData = new FluidData();
+        fluidData.loadLibrary();
+        FluidFactory fluidFactory= new FluidFactory(fluidData);
+        IdealProcess idealProcess = new IdealProcess(fluidFactory);
+        RealProcess realProcess = new RealProcess(idealProcess, fluidFactory);
         Parameter<Pressure> startPointPressure = new Parameter(ParameterType.PRESSURE, NonSI.BAR, 8.0);
         Parameter<Quality> startPointQuality = new Parameter(ParameterType.QUALITY, Dimensionless.UNIT, 1.0);
-        Fluid startPoint = FluidFactory.createFluid(FluidName.R134A, startPointPressure, startPointQuality );
+        Fluid startPoint = fluidFactory.createFluid(FluidName.R134A, startPointPressure, startPointQuality );
         Parameter<Pressure> endPressure = new Parameter(ParameterType.PRESSURE, NonSI.BAR, 20.0);
         Parameter<Efficiency> compressionEfficiency = new Parameter(Dimensionless.UNIT, 1.0);
         //WHEN
-        Fluid idealCompressionEndPoint = RealProcess.idealCompression(startPoint, endPressure);
-        Fluid realCompressionEndPoint = RealProcess.compression(startPoint, endPressure, compressionEfficiency);
+        Fluid idealCompressionEndPoint = realProcess.idealCompression(startPoint, endPressure);
+        Fluid realCompressionEndPoint = realProcess.compression(startPoint, endPressure, compressionEfficiency);
         //THEN
         assertEquals(idealCompressionEndPoint, realCompressionEndPoint);
```

Comment:
I would add there an assertion with real `compression` value that should be returned there. what if you have bug in both `idealCompression` and `compression` methods and they return e.g `null`? the test would pass.